api/types/header/header.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package header
```

Comment:
I would argue yes because these two structs mirror proto messages/interfaces in `api/types/resource.go`, but have none of the proto backing. We could potentially lift these into `api/types` but we'd have to come up with possibly stuttering or weirdly specific names, e.g. `NonGogoResourceHeader`. I know that name is terrible, but that's effectively what we've got here -- these are the non-gogo versions of these structs that already exist but can't be used due to their gogo backing in `api/types`.

Reply:
I see. Thanks for the reasoning, this is definitely better than `NonGogoResourceHeader`.